Fyne error:  Failed to encode GUI code

2022/08/04 00:09:32 Please select a container
2022/08/04 00:09:45 Please select a container
2022/08/04 00:09:47 Please select a container
2022/08/04 00:11:24 // auto-generated
// Code generated by Defyne GUI builder.

package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
	"fyne.io/fyne/v2/app"
)

type gui struct {

}

func newGUI() *gui {
	return &gui{}
}

func (g *gui) makeUI() fyne.CanvasObject {
	g.5 = container.NewHBox(
		container.New(),
		widget.NewSeparator(),
		widget.NewSeparator())


	return g.5}

2022/08/04 00:11:24 Fyne error:  Failed to encode GUI code
2022/08/04 00:11:24   Cause: 22:3: expected ';', found .5 (and 1 more errors)
2022/08/04 00:11:24   At: /home/o/go/pkg/mod/github.com/fyne-io/defyne@v0.0.0-20220628091741-456876da2bcf/internal/guibuilder/main.go:353
/tmp/fynebuilder/main.go:2:1: expected 'package', found 'func'
2022/08/04 00:11:33 // auto-generated
// Code generated by Defyne GUI builder.

package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
	"fyne.io/fyne/v2/app"
)

type gui struct {

}

func newGUI() *gui {
	return &gui{}
}

func (g *gui) makeUI() fyne.CanvasObject {
	g.5 = container.NewHBox(
		container.New(),
		widget.NewSeparator(),
		widget.NewSeparator())


	return g.5}

2022/08/04 00:11:33 Fyne error:  Failed to encode GUI code
2022/08/04 00:11:33   Cause: 22:3: expected ';', found .5 (and 1 more errors)
2022/08/04 00:11:33   At: /home/o/go/pkg/mod/github.com/fyne-io/defyne@v0.0.0-20220628091741-456876da2bcf/internal/guibuilder/main.go:353
/tmp/fynebuilder/main.go:2:1: expected 'package', found 'func'

--- a/internal/guibuilder/main.go
+++ b/internal/guibuilder/main.go
@@ -14,6 +14,7 @@ import (
 
 	"fyne.io/fyne/v2"
 	"fyne.io/fyne/v2/container"
+	"fyne.io/fyne/v2/data/validation"
 	"fyne.io/fyne/v2/dialog"
 	"fyne.io/fyne/v2/layout"
 	"fyne.io/fyne/v2/storage"
@@ -182,6 +183,7 @@ func (b *Builder) buildUI(content fyne.CanvasObject) fyne.CanvasObject {
 
 	widType = widget.NewLabelWithStyle("(None Selected)", fyne.TextAlignCenter, fyne.TextStyle{Bold: true})
 	widName = widget.NewEntry()
+	widName.Validator = validation.NewRegexp("^[a-zA-Z_][a-zA-Z0-9_]*$", "Invalid variable name")
 	paletteList = container.NewVBox()
 	palette := container.NewBorder(container.NewVBox(widType,
 		widget.NewForm(widget.NewFormItem("Variable", widName))), nil, nil, nil,

--- a/internal/guibuilder/main.go
+++ b/internal/guibuilder/main.go
@@ -234,12 +234,16 @@ func varsRequired(obj fyne.CanvasObject) []string {
 		return []string{w.name + " " + class}
 	}
 
-	ret := []string{}
+	var ret []string
 	var objs []fyne.CanvasObject
 	if c, ok := obj.(*fyne.Container); ok {
 		objs = c.Objects
 	} else if c, ok := obj.(*overlayContainer); ok {
 		objs = c.c.Objects
+
+		if c.name != "" {
+			ret = append(ret, c.name+" "+"*fyne.Container")
+		}
 	}
 	for _, w := range objs {
 		ret = append(ret, varsRequired(w)...)

